source/develop/release-management/features/network/autodefine-external-network.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Autodefine External Network
+category: feature
+authors: Ales Musil
+feature_name: Autodefine External Network
+feature_modules: engine
+feature_status: In Development
+---
+
+# Autodefine External Network
+
+## Summary
+
+This feature extends [Provider Physical Network](https://www.ovirt.org/develop/release-management/features/network/provider-physical-network/) 
+adding support for auto-defined external network connected to physical ones. 
+Engine automatically creates an external network based on physical VM network
+with physical connection to it. 
+
+Related patches can be found on [gerrit topic:autodefine](https://gerrit.ovirt.org/#/q/topic:autodefine).
+
+### Owner
+
+- Feature Owner: Ales Musil (amusil)
+- E-mail: <amusil@redhat.com>
+
+### Benefit to oVirt
+
+Autodefine helps reducing administrative tasks needed with OVS switch type in 
+the cluster.
+
+## Prerequisites
+
+Cluster has to be configured with switch type OVS and valid external provider.
+The user needs to make sure that connection to external provider is valid.
+
+### Limitations
+
+* With OVS switch type in cluster, only external networks are allowed to 
+  be attached to the VM interface. 
+
+* Physical network linked to any external network can't be removed. 
+
+* Existing Clusters switch cannot be changed to OVS. Currently there is no way 
+  in the Engine to follow this change across all networks and VM interfaces.
+
+## Detailed description
+
+### Creating of new physical network
+
+When creating new network choose its name. Check the `VM network`. If vlan 
+tagging is needed, check the `Enable VLAN tagging` and put the tag into the 
+field. Please note that auto-define can be triggered only on top of network that
+is not marked as external. 
+
+![Creation of physical network](../../../../images/wiki/auto-define_create.png  "New physical network")
+
+After confirmation if all conditions were met, the external network connected to 
```

Comment:
- Is there no way for the user to prevent this?
- Should the command fail, if AutodefineExternalNetwork fails?
- What would be the flow to prevent this? Add a non-VM network and change to VM network after the network is created?

Reply:
- Is there no way for the user to prevent this?
No.

- Should the command fail, if AutodefineExternalNetwork fails?
No. An audit log with the reason of the failure should be displayed.

- What should happen if the conditions are not met? A log entry? No log entry?
An external network won't be auto created. There is no need to log it.

- What would be the flow to prevent this? Add a non-VM network and change to VM network after the network is created?
Changing the network to vm network or any other changes that causes 'the conditions to be met' should invoke auto define of the external network.

Reply:
* What would be the flow to prevent this? Add a non-VM network and change to VM network after the network is created?
Changing the network to vm network or any other changes that causes 'the conditions to be met' should invoke auto define of the external network.

I think the update flow might be tricky as there are few edge cases. E.g. what to do when the physical network has auto-defined external one and is updated to non-VM? This problem might apply to all networks with link to external one.

Reply:
Updating to vm->non-vm should not do anything (since non-vm network can be a physical network of external one).
Updating non-vm->vm should auto create external network.

The vm/non-vm property in cluster with ovs switch type is confusing, since it actual meaning is auto create external network.

Reply:
I would add the update flow information later as I am not entirely sure how to detect if there is existing autodefined network yet.